src/lib/es2020.intl.d.ts

```diff
@@ -197,6 +197,10 @@ declare namespace Intl {
         ): UnicodeBCP47LocaleIdentifier[];
     };
 
+    interface NumberFormatPart {
+        type: "compact" | "currency" | "decimal" | "exponentInteger" | "exponentMinusSign" | "exponentSeparator" | "fraction" | "group" | "infinity" | "integer" | "literal" | "minusSign" | "nan" | "plusSign" | "percentSign" | "unit" | "unknown";

```

Comment:
I'd restructure this to something like
```ts
// ES2018
type ES2018NumberFormatPartType = "literal" | "nan" | "infinity" | "percent" | "integer" | "group" | "decimal" | "fraction" | "plusSign" | "minusSign" | "percentSign" | "currency" | "code" | "symbol" | "name"
// ES2020
type ES2020NumberFormatPartType = ES2018NumberFormatPartType | "compact" | "exponentInteger" | "exponentMinusSign" | "exponentSeparator" | "unit" | "unknown"

interface NumberFormatPart {
	type: ES2020NumberFormatPartType
	value: string
}
```

Or at least keep the order the same across ES versions so it's easier to review

Reply:
Great ideas, implemented 👍🏻  and pushed